pkg/prometheus/promcfg.go

```diff
@@ -407,9 +412,6 @@ func (cg *ConfigGenerator) buildExternalLabels() yaml.MapSlice {
 		m[replicaExternalLabelName] = fmt.Sprintf("$(%s)", operator.PodNameEnvVar)
 	}
 
-	for n, v := range cpf.ExternalLabels {
```

Comment:
I did added a check initially but wasn't sure how users use it currently if that will break their logic (even though api says it will be overridded) :)

But I think its best to skip reserved ones (default names only). wdyt?